api/tests/integration/infrastructure/repositories/campaign-repository_test.js

```diff
@@ -136,33 +137,65 @@ describe('Integration | Repository | Campaign', () => {
       organizationId: organizationId,
       creatorId: 2
     });
-    const campaign1Organization2 = domainBuilder.buildCampaign({
+
+    const campaignOrganization1WithoutParticipant = domainBuilder.buildCampaign({
       id: 3,
       name: 'campaign3',
       code: 'AZERTY789',
-      organizationId: 2,
+      organizationId: organizationId,
       creatorId: 3
     });
-    const campaigns = [campaign1Organization1, campaign2Organization1, campaign1Organization2];
+    const campaigns = [campaign1Organization1, campaign2Organization1, campaignOrganization1WithoutParticipant];
+
+    const campaignParticipation1 = domainBuilder.buildCampaignParticipation({
+      id: 1,
+      campaign: campaign1Organization1,
+      isShared: true
+    });
+    const campaignParticipation2 = domainBuilder.buildCampaignParticipation({
+      id: 2,
+      campaign: campaign1Organization1,
+      isShared: false
+    });
+    const campaignParticipation3 = domainBuilder.buildCampaignParticipation({
+      id: 3,
+      campaign: campaign1Organization1,
+      isShared: false
+    });
+    const campaignParticipation4 = domainBuilder.buildCampaignParticipation({
+      id: 4,
+      campaign: campaign2Organization1,
+      isShared: true
+    });
+    const campaignParticipation5 = domainBuilder.buildCampaignParticipation({
+      id: 5,
+      campaign: campaign2Organization1,
+      isShared: true
+    });
+    const campaignParticipations = [campaignParticipation1, campaignParticipation2, campaignParticipation3, campaignParticipation4, campaignParticipation5];
```

Comment:
J'aurai fais pas mal de choses différemment ici, sur la base de ce qui a déjà été fait dans d'autres PR. 
- Pas utiliser d'objets du `domain`, ça n'apporte rien pour les injecter en base à par du bruit. On ne s'en sert que pour vérifier que ce qui sort du repo est bien un objet du domaine. 
- Pas d'assignation à des variables si on ne les utilise qu'une fois, ça rajoute également du bruit.
- Utilisation du `members` (cf. commentaire précédent) pour éviter les `...[0]...` à rallonge dans les expect et s'affranchir de l'ordre !
- Ajout d'une campagne supplémentaire reliée à une autre organisation pour tester le filtre
- `async/await` pour faire comme dans le test du `get` et parce que c'est plus joli
- Tout mettre dans un seul `it` en mutualisant l'appel dans un `beforeEach`
- Créer un objet du domaine pour le `CampaignReport` même s'il est tout petit je pense qu'il en faut un !
- Ne pas teste tous les champs du premier objet. La fonction `_fromRawToDomain`devrait être testé entièrement et une fois à travers le `get` par exemple pour ne pas ajouter du bruit aux autres tests

Ca donnerait: 

```
const organizationId = 1;
    let campaignsWithReports;

    beforeEach(async () => {
      _.each([
        { id: 1, name: 'campaign1', code: 'AZERTY123', organizationId, creatorId: 1 },
        { id: 2, name: 'campaign2', code: 'AZERTY456', organizationId, creatorId: 2 },
        { id: 3, name: 'campaign without participations', code: 'AZERTY789', organizationId, creatorId: 3 },
        { id: 4, name: 'campaign without participations', code: 'AZERTY789', organizationId: 4, creatorId: 3 },

      ], (campaign) => {
        databaseBuilder.factory.buildCampaign(campaign);
      });

      _.each([
        { id: 1, campaignId: 1, isShared: true },
        { id: 2, campaignId: 1, isShared: false },
        { id: 3, campaignId: 1, isShared: false },
        { id: 4, campaignId: 2, isShared: true },
        { id: 5, campaignId: 2, isShared: true },

      ], (campaignParticipation) => {
        databaseBuilder.factory.buildCampaignParticipation(campaignParticipation);
      });

      await databaseBuilder.commit();

      campaignsWithReports = await campaignRepository.findByOrganizationIdWithCampaignReports(organizationId);
    });

    afterEach(async () => {
      await databaseBuilder.clean();
    });

    it.only('should return the campaigns of the given organization id with campaignReports', async () => {
      expect(_.map(campaignsWithReports, 'id')).to.have.members([1, 2, 3]);
      expect(campaignsWithReports[0]).to.be.instanceOf(Campaign);
      // expect(campaignsWithReports[0].campaignReport).to.be.instanceOf(CampaignReport);
      expect(campaignsWithReports[0].campaignReport).to.deep.equal({ id: 1, participationsCount: 3, sharedParticipationsCount: 1 });
      expect(campaignsWithReports[1].campaignReport).to.deep.equal({ id: 2, participationsCount: 2, sharedParticipationsCount: 2 });
      expect(campaignsWithReports[2].campaignReport).to.deep.equal({ id: 3, participationsCount: 0, sharedParticipationsCount: 0 });

    });
```

Reply:
C'est beau